icechunk/src/repository.rs

```diff
@@ -355,8 +355,20 @@ impl Repository {
     /// Deletes of non existing groups will succeed.
     pub async fn delete_group(&mut self, path: Path) -> RepositoryResult<()> {
         match self.get_group(&path).await {
-            Ok(node) => {
-                self.change_set.delete_group(node.path, &node.id);
+            Ok(parent) => {
+                let nodes_iter: Vec<NodeSnapshot> = self.list_nodes().await?.collect();
```

Comment:
How about calling `filter` on the result of `list_nodes` passing `is_prefix_match`. That way, you can construct a much smaller `Vec`